autoPyTorch/pipeline/components/training/trainer/base_trainer_choice.py

```diff
@@ -0,0 +1,479 @@
+        super().check_requirements(X, y)
+
+        # We need a working dir in where to put our data
+        if 'working_dir' not in X:
+            raise ValueError('Need a working directory to output trainer information, '
+                             "yet 'working_dir' was not found in the fit dictionary")
+
+        # For resource allocation, we need to know if pynisher is enabled
+        if 'use_pynisher' not in X:
+            raise ValueError('Missing use_pynisher in the fit dictionary')
+
+        # Whether we should evaluate metrics during training or no
+        if 'metrics_during_training' not in X:
+            raise ValueError('Missing metrics_during_training in the fit dictionary')
+
+        # Setup Components
+        if 'lr_scheduler' not in X:
+            raise ValueError("Learning rate scheduler not found in the fit dictionary!")
+
+        if 'network' not in X:
+            raise ValueError("Network not found in the fit dictionary!")
+
+        if 'optimizer' not in X:
+            raise ValueError("Optimizer not found in the fit dictionary!")
+
+        # Training Components
+        if 'train_data_loader' not in X:
+            raise ValueError("train_data_loader not found in the fit dictionary!")
+
+        if 'val_data_loader' not in X:
+            raise ValueError("val_data_loader not found in the fit dictionary!")
+
+        if 'budget_type' not in X:
+            raise ValueError("Budget type not found in the fit dictionary!")
+        else:
+            if 'epochs' not in X or 'runtime' not in X or 'epoch_or_time' not in X:
+                if X['budget_type'] in ['epochs', 'epoch_or_time'] and 'epochs' not in X:
+                    raise ValueError("Budget type is epochs but "
+                                     "no epochs was not found in the fit dictionary!")
+                elif X['budget_type'] in ['runtime', 'epoch_or_time'] and 'runtime' not in X:
+                    raise ValueError("Budget type is runtime but "
+                                     "no maximum number of seconds was provided!")
+            else:
+                raise ValueError("Unsupported budget type provided: {}".format(
+                    X['budget_type']
+                ))
+
+        if 'job_id' not in X:
+            raise ValueError('Need a job identifier to be able to isolate jobs')
+
+        for config_option in ["torch_num_threads", 'device']:
+            if config_option not in X:
+                raise ValueError("Missing config option {} in config".format(
+                    config_option
+                ))
+
+    def get_device(self, X: Dict[str, Any]) -> torch.device:
+        """
+        Returns the device to do torch operations
+
+        Args:
+            X (Dict[str, Any]): A fit dictionary to control how the pipeline
+                is fitted
+
+        Returns:
+            torch.device: the device in which to compute operations. Cuda/cpu
+        """
+        if not torch.cuda.is_available():
```

Comment:
Could we add a logging info here that CPU or GPU is chosen?

Reply:
I will add this in next push!